rancher-sandbox/rancher-desktop
# Feat/containers-namespaces

Fixes https://github.com/rancher-sandbox/rancher-desktop/issues/5714

What's new:
- While on `containerd`, namespaces are shown (same way as Images).
- actions (start, stop...) will be run with a `nerdctl` command  
This prepares the `backend` to handle `containers` but in a generic way (due to time constraints) 


## How to test:
1. Make sure you switch to `containerd`.
2. Run `nerdctl run -d -p 5100:1333 redis` 
3. You should see the container under the **Containers** page
4. You should be able to **Start/Stop/Remove** the Container.
5. You should be able to **switch namespaces**


NOTE: https://github.com/containerd/nerdctl/issues/665 This happens sometimes, while looking for it I noticed that @jandubois has been there and maybe you have some contexts, I am not sure if there's something from the UI we can do. It will fail on the first START but, it will work the next time you click to start the container.

## Screenshoots
![image](https://github.com/rancher-sandbox/rancher-desktop/assets/88777903/ca0599e4-18d2-4925-8cba-b9e87bb91e58)
